lightning/src/onion_message/messenger.rs

```diff
@@ -595,6 +615,9 @@ pub trait CustomOnionMessageHandler {
 	/// The returned [`Self::CustomMessage`], if any, is enqueued to be sent by [`OnionMessenger`].
 	fn handle_custom_message(&self, message: Self::CustomMessage, responder: Option<Responder>) -> ResponseInstruction<Self::CustomMessage>;
 
+	#[cfg(test)]
+	fn test_handle_custom_message(&self, message: Self::CustomMessage, responder: Option<Responder>, add_reply_path: bool) -> ResponseInstruction<Self::CustomMessage>;
+
```

Comment:
You shouldn't need to expand the trait, even if it is test-only. Instead, either make a separate implementation that uses a different `ResponseInstruction` or modify the implementation to reply differently based some field, for instance. Then have the test set the field as needed.

A more general approach is to give the test implementation a set of expectations/actions (e.g., expect that the parameter to the method is X and return Y). For examples, take a look at `TestRouter`, `TestScorer`, `TestKeysInterface`, and `TestChainSource` in https://github.com/lightningdevkit/rust-lightning/blob/main/lightning/src/util/test_utils.rs. They all utilize this approach in slightly different ways.

Reply:
Thanks a lot for the suggestion and the references!

In [5e99d68](https://github.com/lightningdevkit/rust-lightning/commit/5e99d68c383addc464f24bd6049c8aaac73e99c2):
Instead of creating a new test-specific `handle_custom_message`, I have utilized the newly introduced `TestCustomMessage` variants to allow the `respond_with_reply_path` test.
This allowed me to leverage the `expect_message` setup too for the `TestCustomMessageHandler`.

Also, I have expanded the variants docs to explain this behavior properly.
Let me know if this approach seems sound!
Thanks a lot!